nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/canvas/ui/connection/source/source-process-group/source-process-group.component.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, forwardRef, Input } from '@angular/core';
+import { ControlValueAccessor, FormsModule, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { NgForOf, NgIf } from '@angular/common';
+import { TextTip } from '../../../../../../ui/common/tooltips/text-tip/text-tip.component';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatOptionModule } from '@angular/material/core';
+import { MatSelectModule } from '@angular/material/select';
+import { NifiTooltipDirective } from '../../../../../../ui/common/tooltips/nifi-tooltip.directive';
+import { SelectOption, TextTipInput } from '../../../../../../state/shared';
+
+@Component({
+    selector: 'source-process-group',
+    standalone: true,
+    templateUrl: './source-process-group.component.html',
+    styleUrls: ['./source-process-group.component.scss'],
+    imports: [
+        MatCheckboxModule,
+        NgForOf,
+        NgIf,
+        FormsModule,
+        MatFormFieldModule,
+        MatOptionModule,
+        MatSelectModule,
+        NifiTooltipDirective
+    ],
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => SourceProcessGroup),
+            multi: true
+        }
+    ]
+})
+export class SourceProcessGroup implements ControlValueAccessor {
+    @Input() set processGroup(processGroup: any) {
+        if (processGroup) {
+            if (processGroup.permissions.canRead) {
+                this.groupName = processGroup.component.name;
+            } else {
+                this.groupName = processGroup.id;
+            }
+        }
+    }
+
+    @Input() set outputPorts(outputPorts: any[]) {
+        if (outputPorts) {
+            this.noPorts = outputPorts.length == 0;
+            this.hasUnauthorizedPorts = false;
+
```

Comment:
I had to solve this slightly differently because the `Input` setter ran before the `ControlValueAccessor` lifecycle hook which set `onTouched`. Please have a look at my latest commit.